### PR TITLE
Make spectrum x-axis tick labels invisible by default in `plot_fit`

### LINKED ISSUE #5339: Hide spectrum x-axis tick labels in `plot_fit`
In `FluxPointsDataset.plot_fit()` and `PlotMixin.plot_fit()` the label of the x-axis tick of `ax_spectrum` is visible behind the residuals plot (see, e.g. https://docs.gammapy.org/1.2/tutorials/analysis-1d/spectral_analysis.html)

I propose to make the x-axis tick labels invisible when `ax_spectrum` and `ax_residuals` are not introduced to `plot_fit`. I think I found the solution in both cases. I can open a PR.

### PR DESCRIPTION
**Pull request by aaguasca**
**Description**

This pull request resolves #5339

**Dear reviewer**

This PR is ready for review.

### CODE REVIEW

File: gammapy/datasets/spectrum.py [hunk 1]
```diff
@@ -58,6 +58,10 @@ def plot_fit(
         >>> dataset.plot_fit(kwargs_residuals=kwargs_residuals, kwargs_spectrum=kwargs_spectrum)  # doctest: +SKIP
         """
         gs = GridSpec(7, 1)
+        if ax_spectrum is None and ax_residuals is None:
+            bool_visible_xticklabel = False
+        else:
+            bool_visible_xticklabel = True
```
Comment (VascoSch92) [inline gammapy/datasets/spectrum.py:64]:
```suggestion
        bool_visible_xticklabel = not ax_spectrum is None and ax_residuals is None
```

Comment (Astro-Kirsty) [reply gammapy/datasets/spectrum.py:64]: Yes but this is missing a bracket I believe:

`bool_visible_xticklabel = not (ax_spectrum is None and ax_residuals is None)`

Comment (VascoSch92) [reply gammapy/datasets/spectrum.py:64]: yes you are right

Comment (aaguasca) [reply gammapy/datasets/spectrum.py]: Thank you for the suggestion! I modified the code accordingly.

**Review by QRemy — APPROVED** [review]
Thanks @aaguasca, no further comment, merging now.

**Pull request merged**